src/main/scala/mesosphere/marathon/MarathonModule.scala

```diff
@@ -101,6 +110,17 @@ class MarathonModule(conf: MarathonConf, http: HttpConf, zk: ZooKeeperClient)
     }
   }
 
+  @Provides
+  @Singleton
+  def provideLeadershipCallbacks(
```

Comment:
This is missing the TaskStore. When using an EntityStoreCache one could enforce that the Cache is leadership aware – you could automate this :)